pkg/cdi/cache.go

```diff
@@ -238,6 +315,11 @@ func (c *Cache) ListClasses() []string {
 	c.Lock()
 	defer c.Unlock()
 
+	// in self-refresh mode trigger a refresh here if a Spec dir appears
+	if c.selfRefresh && c.watch.update(c.dirErrors) {
```

Comment:
Question: Since we're using this regularly now, does it make sense to wrapt this in a method `mustRefresh`? or even `refreshIfRequired()` to improve readability and remove the need to keep repeating this comment?

Reply:
Good point and also much less error-prone for future changes. Updated accordingly.